xla/tests/collective_ops_test_e2e.cc

```diff
@@ -855,5 +876,98 @@ ENTRY main.9_spmd {
+  add.232 = s32[] add(subtract.139, constant.2562)
+  select.1348 = s32[] select(compare.747, add.232, add.231)
+  dynamic-slice.k = bf16[32,128] dynamic-slice(get-tuple-element.k, select.1348, constant.2561), dynamic_slice_sizes={32,128}
+  r = bf16[32,128] bitcast(dynamic-slice.k)
+  a = bf16[32,128] add(r, r), control-predecessors={constant.2559}
+  // A fp8 pattern of quant-dequant before the collective AG.
+  qa = f8e4m3fn[32,128] convert(a)
+  dqa = bf16[32,128] convert(qa)
+  a_scale = bf16[] get-tuple-element(param), index=3
+  a_scales = bf16[32,128] broadcast(a_scale), dimensions={}
+  dqa_unscaled = bf16[32,128] multiply(dqa, a_scales)
+  mb = bf16[128,128] all-gather(dqa_unscaled), channel_id=1, use_global_device_ids=true, dimensions={0}, replica_groups={{0,1,2,3}}
+  ma = bf16[128,128] dynamic-slice(get-tuple-element.395, select.1348, constant.2561), dynamic_slice_sizes={128,128}
+  
+  qma = f8e4m3fn[128,128] convert(ma)
+  dqma = bf16[128,128] convert(qma)
+  ma_scale = bf16[] get-tuple-element(param), index=4
+  ma_scales = bf16[128,128] broadcast(ma_scale), dimensions={}
+  dqma_unscaled = bf16[128,128] multiply(dqma, ma_scales)
+  mc = bf16[128,128] dot(dqma_unscaled, mb), lhs_contracting_dims={1}, rhs_contracting_dims={0}
+  dynamic-update-slice.35 = bf16[384,128] dynamic-update-slice(get-tuple-element.395, mc, select.1348, constant.2561)
+  ROOT tuple = (s32[], bf16[384,128], bf16[96,128], bf16[], bf16[]) tuple(add.230, dynamic-update-slice.35, get-tuple-element.k, a_scale, ma_scale), control-predecessors={a}
+}
+ENTRY entry {
+  c0 = s32[] constant(0)
+  p0 = bf16[384,128] parameter(0)
+  p1 = bf16[96,128] parameter(1)
+  s0 = bf16[] parameter(2)
+  s1 = bf16[] parameter(3)
+  tuple = (s32[], bf16[384,128], bf16[96,128], bf16[], bf16[]) tuple(c0, p0, p1, s0, s1)
+  while = (s32[], bf16[384,128], bf16[96,128], bf16[], bf16[]) while(tuple), condition=while_cond, body=while_body
+  ROOT gte1 = bf16[384,128] get-tuple-element(while), index=1
+}
+)";
+
+  const int64_t kNumReplicas = 1;
+  const int64_t kNumPartitions = 4;
+
+  HloModuleConfig config =
+      GetModuleConfigForTest(/*replica_count=*/kNumReplicas);
+  auto opts = GetDebugOptionsForTest();
+  opts.set_xla_gpu_run_post_layout_collective_pipeliner(true);
+  opts.set_xla_gpu_enable_pipelined_collectives(true);
+  opts.set_xla_gpu_enable_triton_gemm(false);
+  config.set_debug_options(opts);
+  config.set_num_partitions(kNumPartitions);
+  TF_ASSERT_OK_AND_ASSIGN(
+      auto module, ParseAndReturnVerifiedModule(kModuleReplicatedStr, config));
+
+  TF_ASSERT_OK_AND_ASSIGN(auto executable,
+                          CreateExecutable(std::move(module),
+                                           /*run_hlo_passes=*/true));
+  EXPECT_TRUE(executable->has_module());
+  HloInstruction* gemm_op =
+      FindInstruction(&executable->module(), HloOpcode::kCustomCall);
+  EXPECT_THAT(gemm_op, NotNull());
+  EXPECT_TRUE(gemm_op->custom_call_target() == "__cublas$lt$matmul$f8");
```

Comment:
Can you please use EXPECT_EQ(gemm_op->custom_call_target(), "__cublas$lt$matmul$f8") ?
The linter complains about this line.